clippy_lints/src/loops.rs

```diff
@@ -1153,27 +1153,70 @@ fn detect_same_item_push<'tcx>(
                     let vec_str = snippet_with_macro_callsite(cx, vec.span, "");
                     let item_str = snippet_with_macro_callsite(cx, pushed_item.span, "");
                     if let ExprKind::Path(ref qpath) = pushed_item.kind {
-                        if_chain! {
```

Comment:
We can move the check in line 1152 (sorry github don't allow commenting on non-edited code) to the beginning of the function, so we can avoid doing some work if not necessary

```rust
    // Make sure that the push does not involve possibly mutating values
    if !matches!(pat.kind, PatKind::Wild) {
        return;
    }
```